tests/test_utils.py

```diff
@@ -899,10 +903,170 @@ def test_companion_mass_error_noquantity_Pb():
+    # but recalculated based on the values above
+    assert np.isclose(PBDOT(Mp, Mc, Pb, e), -1.24777223e-12 * u.s / u.s)
+
+
+def test_gamma_error_noquantity_mp():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        GAMMA(Mp.value, Mc, Pb, e)
+
+
+def test_gamma_error_noquantity_mc():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        GAMMA(Mp, Mc.value, Pb, e)
+
+
+def test_gamma_error_noquantity_pb():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        GAMMA(Mp, Mc, Pb.value, e)
+
+
+def test_omdot_error_wrongquantity_mp():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(u.UnitsError):
+        OMDOT(Mp.value * u.s, Mc, Pb, e)
+
+
+def test_omdot_error_wrongquantity_mc():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(u.UnitsError):
+        OMDOT(Mp, Mc.value * u.s, Pb, e)
+
+
+def test_omdot_error_wrongquantity_pb():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(u.UnitsError):
+        OMDOT(Mp, Mc, Pb.value * u.Msun, e)
+
+
+def test_omdot_error_wrongquantity_e():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(u.UnitsError):
+        OMDOT(Mp, Mc, Pb, e * u.s)
+
+
+def test_omdot_error_noquantity_mp():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        OMDOT(Mp.value, Mc, Pb, e)
+
+
+def test_omdot_error_noquantity_mc():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        OMDOT(Mp, Mc.value, Pb, e)
+
+
+def test_omdot_error_noquantity_pb():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        OMDOT(Mp, Mc, Pb.value, e)
+
+
+def test_pbdot_error_noquantity_mp():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        PBDOT(Mp.value, Mc, Pb, e)
+
+
+def test_pbdot_error_noquantity_mc():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        PBDOT(Mp, Mc.value, Pb, e)
+
+
+def test_pbdot_error_noquantity_pb():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        PBDOT(Mp, Mc, Pb.value, e)
+
+
+def test_A1SINI():
```

Comment:
Is it worth this being a parametrized calculation? Like, it's a round-trip check, so it should work for any reasonable combination of input values. With `pytest.mark.parametrize` you could quickly check three or four such combinations; with `hypothesis` you could make it try lots of combinations (but it doesn't directly support units and is really aggressive about finding failures).

Reply:
Sure.  I added in `hypothesis`-based tests.